tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx

```diff
@@ -137,42 +160,104 @@ public:
    RNTupleInspector &operator=(RNTupleInspector &&other) = delete;
    ~RNTupleInspector() = default;
 
-   /// Create a new inspector for a given RNTuple. When this factory method is called, all required static information
-   /// is collected from the RNTuple's fields and underlying columns are collected at ones. This means that when any
-   /// inconsistencies are encountered (e.g. inconsistent compression across clusters), it will throw an error here.
-   static std::unique_ptr<RNTupleInspector> Create(std::unique_ptr<Detail::RPageSource> pageSource);
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Create a new RNTupleInspector.
+   ///
+   /// \param[in] sourceNTuple A pointer to the RNTuple to be inspected.
+   ///
+   /// \return A pointer to the newly created RNTupleInspector.
+   ///
+   /// \note When this factory method is called, all required static information is collected from the RNTuple's fields
+   /// and underlying columns are collected at ones. This means that when any inconsistencies are encountered (e.g.
+   /// inconsistent compression across clusters), it will throw an error here.
    static std::unique_ptr<RNTupleInspector> Create(RNTuple *sourceNTuple);
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Create a new RNTupleInspector.
+   ///
+   /// \param[in] ntupleName The name of the RNTuple to be inspected.
+   /// \param[in] storage The path or URI to the RNTuple to be inspected.
+   ///
+   /// \see Create(RNTuple *sourceNTuple)
    static std::unique_ptr<RNTupleInspector> Create(std::string_view ntupleName, std::string_view storage);
 
-   /// Get the descriptor for the RNTuple being inspected.
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Create a new RNTupleInspector.
+   ///
+   /// \param[in] pageSource The RPageSource object belonging to the RNTuple to be inspected.
+   ///
+   /// \see Create(RNTuple *sourceNTuple)
+   static std::unique_ptr<RNTupleInspector> Create(std::unique_ptr<Detail::RPageSource> pageSource);
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get the descriptor for the RNTuple being inspected.
+   ///
+   /// \return A static copy of the RNTupleDescriptor belonging to the inspected RNTuple.
```

Comment:
Does this return a "static copy"?

Reply:
`fDescriptor` holds a clone of the `RNTuple` that is being inspected:
https://github.com/root-project/root/blob/1a0a84806ebca04ce794db0581b054371aaac2b8/tree/ntupleutil/v7/src/RNTupleInspector.cxx#L35-L37
So it's a static copy in the sense that if for any reason the inspected `RNTuple` changes during the lifetime of the inspector object, these changes won't be reflected there. I think the word 'static' may cause confusion here though, so I'll remove it.